### ISSUE TITLE
More a question, than an issue

### ISSUE THREAD

**Issue by bobogumizelka** [iArunava/ENet-Real-Time-Semantic-Segmentation#11]
Hi Arunava,

I find your work very interesting, because I need to use segmentation on nvidia Jetson - and popular models are too slow for me. I red your article and the original paper, and I can't find the information about the number of epochs needed to train model with good performance. I used your notebook to train pedestrian segmentation (2 classes - pedestrian and background). After 100 epochs model couldn't segment anything at all - empty masks. Then I used your pretrained weights and fine-tuned the model for 2 classes, 100 epochs again. This time I got segmentation masks, but they were completely inaccurate. How long should I train Enet to get accurate results? Or maybe I should already get some reasonable results, and it seems I'm doing something wrong (perhaps with data)?  
Could you also give me some hint about the losses that are printed during training? What are the sensible values? I'm getting very big number or sometimes negatives, I feel like something is wrong.

**Comment by AvivSham** [comment]
Can you please describe your preprocess part? maybe something will come up.
I also don't know what is the presence of "pedestrian" in your data samples. BTW if you don't use the same dataset as we did some modifications are needed.

**Comment by bobogumizelka** [comment]
I made some modifications - after loading your pretrained weights I change last layer: enet.fullconv = nn.ConvTranspose2d(16, 2, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), output_padding=(1, 1), bias=False) for 2 classes (pedestrian/background).
I use PenFudanPed dataset: https://www.cis.upenn.edu/~jshi/ped_html/, I changed instance segmentation masks into semantic segmentation mask (0 for background, 1 for pedestrian).

**Comment by AvivSham** [comment]
please describe you preprocess part please.

**Comment by bobogumizelka** [comment]
I don't do any special preprocessing. I use loader function from your notebook. I load images (png), resize them to 1000x320, change image format to C x H x W. I load masks (png) with 0 for background, and 1, 2, 3, ... for pedestrians, then I change every non-zero value to 1, to get semantic segmentation mask, resize to 1000x320. 

Should I use some special normalization before? CAMVID dataset was downloaded and used right away, I guess?

**Comment by AvivSham** [comment]
What is the resolution of the dataset you are using?

**Comment by bobogumizelka** [comment]
Pictures have various resolution, from 53x323 to 1017x444 px. So I shouldn't resize them all to fixed resolution, or keep ratio aspect? But I guess Enet needs fixed size of images for training?

**Comment by AvivSham** [comment]
This can be the problem! remember the model was trained on common data not real life data we all know that in real life data the problems relate to the data. Please try to resize them to the size of CAMVID dataset and double check the quality of the images if they are fine go ahead and train the model. 
** Be aware that you will need to resize the labels ass well since the labels are indexed images please use nearest neighbor method to resize them

**Comment by bobogumizelka** [comment]
Thanks very much  for your answers and advice! I will check it out.

**Comment by AvivSham** [comment]
I'm happy to help :)
